CHANGELOG.next.toml

```diff
@@ -225,3 +225,45 @@ message = "Support @deprecated trait for aggregate shapes"
 references = ["smithy-rs#1570"]
 meta = { "breaking" = true, "tada" = true, "bug" = false }
 author = "weihanglo"
+
+[[aws-sdk-rust]]
+message = "`aws_config::RetryConfig` no longer implements `Default`, and its `new` function has been replaced with `standard`."
+references = ["smithy-rs#1603", "aws-sdk-rust#586"]
+meta = { "breaking" = true, "tada" = false, "bug" = false }
+author = "jdisanti"
+
+[[smithy-rs]]
+message = "`aws_smithy_types::RetryConfig` no longer implements `Default`, and its `new` function has been replaced with `standard`."
+references = ["smithy-rs#1603", "aws-sdk-rust#586"]
+meta = { "breaking" = true, "tada" = false, "bug" = false, "target" = "client" }
+author = "jdisanti"
+
+[[aws-sdk-rust]]
+message = """
+Direct configuration of `aws_config::SdkConfig` now defaults to retries being disabled.
+If you're using `aws_config::load_from_env()` or `aws_config::from_env()` to configure
+the SDK, then you are NOT affected by this change. If you use `SdkConfig::builder()` to
+configure the SDK, then you ARE affected by this change and should set the retry config
+on that builder.
+"""
+references = ["smithy-rs#1603", "aws-sdk-rust#586"]
+meta = { "breaking" = true, "tada" = false, "bug" = false }
+author = "jdisanti"
+
+[[smithy-rs]]
+message = "Clients now default to retry being disabled if no `RetryConfig` is given to the `Config`"
```

Comment:
I'm unsure about this change, which is why I have this PR set to Draft. I think we need the SDK client behavior to diverge from the non-SDK client behavior here, since the SDK can rely on `aws-config` to get default retry config, and non-SDK clients can't.

Reply:
From discussion with @rcoh, we should default to retry being on for manually constructed SDK clients, and panic if the sleep impl isn't provided.